File: Divisor.py
import psycopg2
def  post(x,y):
    try:
        conexion = psycopg2.connect(
            host = "localhost",
            port = "5432",
            user = "postgres",
            password = "123fgthg",
            dbname = "HWP"
            )
    except:
        print("Sin Conexion Exitosa\n")
            
    cursor = conexion.cursor()
    cursor.execute("INSERT INTO Divisores(numero,divisores) VALUES (%s, %s  );", (x, y))
    conexion.commit()
    cursor.close()
    conexion.close()

print("Divisores de un numero\n")
print("A continuacion ingrese su numero:\n")
a = int(input("Ingrese su numero \n"))
r=[]
z=1
for z in range(1,a+1):
    b = a%z
    if b == 0:
        r.append(z)

print("Los Divisores de "+str(a)+" son: "+str(r))
post(a,r)
file = open("C:/Users/Roberto/Documents/Proyectos/TAREA 1/Divisor.txt","w")
file.write("Los Divisores de "+str(a)+" son: "+str(r))
file.close()
exit(0)

File: Divisor.m
clc; clear all; close all;
pkg load database;
fprintf('Bienvenido al programa\n');
fprintf('Divisores de un numero: A continuacion ingrese un numero\n');
a=input('Ingrese su numero:');
c=1;
for x=1:a
  y=mod(a,x);
  if y==0
    V(c)=x;
    c=c+1;
    
  endif
endfor
fprintf('Los divisores de %i son:\n', a);
disp(V);
s=mat2str(V);
params= cell(1,2);
params{1,1}=a;
params{1,2}=s;
save('Divisor.txt','V','-ascii');
conn = pq_connect(setdbopts('dbname','HWP','host','localhost',
'port','5432','user','postgres','password','123fgthg'));
N=pq_exec_params(conn, "insert into Divisores(numero,divisores) values($1,$2);",params); %insertar datos en la tabla
